run_experiment.py

```diff
@@ -122,19 +131,39 @@ def parse_arguments():
     dir_opts.add_argument('-o', '--output_file_name', help='The full path name to use when saving the plot.')
 
     args = parser.parse_args()
+
+    if args.log_dir is None:
+        args.log_dir = args.model_dir + 'losses.log'
+
     return args
 
 
 def main(args):
     """ The primary entry point for carrying out experiments on pretrained models.
     """ 
-    experiment_name = args.experiment
+    formatter = logging.Formatter('%(asctime)s %(levelname)s - %(funcName)s: %(message)s',
+                                  "%H:%M:%S")
+    logger = logging.getLogger(__name__)
+    logger.setLevel(args.log_level.upper())
+    stream = logging.StreamHandler()
+    stream.setLevel(args.log_level.upper())
+    stream.setFormatter(formatter)
+    logger.addHandler(stream)
+
+    experiment_name = args.model_dir
     dataset = read_dataset_from_specs('experiments/{}/specs.json'.format(experiment_name))
 
     if args.visualisation == 'snapshot_recon':
         viz_list = []
         epoch_list = []
-        model_list = load_model(directory='experiments/{}'.format(experiment_name), load_snapshots=True)
+
+        model_list = load_model(
+            directory='experiments/{}'.format(experiment_name),
+            load_snapshots=True
```

Comment:
I changed this to `load_checkpoints` (sorry I actually didn't test it but I'd be srprised if it didn't work).

using if statements to change the return of a function is very bad practice. Again when you see `load_model` do you think that it can either return a list or a single model depedning on the params ? 

Python is great for soooo many things but it let's you code very dirtily. When easy it's always better to code c++ style, where teh type of the return is known. I always think about my code as being an API that everyone can just call without reading the code.